graph-oauth/src/auth.rs

```diff
@@ -364,6 +364,21 @@ impl OAuth {
         self.insert(OAuthCredential::RefreshTokenURL, value)
     }
 
+    /// Set both the access token and refresh token of a request
+    /// for OAuth based on a tenant id.
+    /// 
+    /// # Example
+    /// ```
+    /// # use graph_oauth::oauth::OAuth;
+    /// # let mut oauth = OAuth::new();
+    /// oauth.tenant_id("tenant_id");
+    /// ```
+    pub fn tenant_id(&mut self, value: &str) -> &mut OAuth {
+        let token_url = format!("https://login.microsoftonline.com/{}/oauth2/v2.0/token", value);
+        self.access_token_url(&token_url)
+            .refresh_token_url(&token_url)
```

Comment:
Everything looks good here. We should also set the authorization URL as well since the path is different from the token endpoints. 

```suggestion
        let token_url = format!("https://login.microsoftonline.com/{}/oauth2/v2.0/token", value);
        let auth_url = format!("https://login.microsoftonline.com/{}/oauth2/v2.0/authorize", value);
        
        self.authorize_url(&auth_url)
            .access_token_url(&token_url)
            .refresh_token_url(&token_url)
```